src/SIS_dyn_cgrid.F90

```diff
@@ -1737,17 +1748,183 @@ subroutine basal_stress_coeff_C(G, mi, ci, sea_lev, CS)
       else
         CS%Tb_v(i,J) = 0.0
       endif
-!     if (G%idg_offset + i == 471 .and. G%jdg_offset + J == 671) then
-!     if (G%idg_offset + i == 25 .and. G%jdg_offset + J == 573) then
-!       print *, 'inside basal_stress', hw_v, ci_v, h_v, hc_v
-!     endif
     enddo
   enddo
 !          call uvchksum("Tb_[uv] before SIS_C_dynamics", CS%Tb_u, CS%Tb_v, G, &
 !                         halos=1, scale=US%RZ_T_to_kg_m2s*US%L_T_to_m_s)
 
 end subroutine basal_stress_coeff_C
 
+!> Computes basal stress Tbu coefficients (landfast ice)
+!!
+!! Computes seabed (basal) stress factor Tbu (landfast ice) based on
+!! probability of contact between the ITD and the seabed. The water depth
+!! could take into account variations of the SSH. In the simplest
+!! formulation, hwater is simply the value of the bathymetry. To calculate
+!! the probability of contact, it is assumed that the bathymetry follows
+!! a normal distribution with sigma_b = 2.5d0. An improvement would
+!! be to provide the distribution based on high resolution data.
+!!
+!! Dupont, F. Dumont, D., Lemieux, J.F., Dumas-Lefebvre, E., Caya, A.
+!! in prep.
+!!
+!! authors: D. Dumont, J.F. Lemieux, E. Dumas-Lefebvre, F. Dupont
+!!
+subroutine basal_stress_coeff_itd(G, IG, IST, sea_lev, CS)
+
+  type(SIS_hor_grid_type),            intent(in)  :: G     !< The horizontal grid type
+  type(ice_grid_type),                intent(in)  :: IG    !< The sea-ice specific grid type
+  type(ice_state_type),               intent(in)  :: IST   !< A type describing the state of the sea ice
+  real, dimension(SZI_(G),SZJ_(G)),   intent(in)  :: sea_lev !< Sea level [Z ~> m]
+  type(SIS_C_dyn_CS),                 pointer     :: CS    !< The control structure for this module
+
+  integer, parameter :: &
+           ncat_b = 100, &  ! number of bathymetry categories
+           ncat_i = 100     ! number of ice thickness categories (log-normal)
+
+  real, parameter :: &
+           max_depth = 50.0, & ! initial range of log-normal distribution (m)
+           mu_s = 0.1          ! friction coefficient
+
+  real, dimension(ncat_i) :: & ! log-normal for ice thickness
+           x_k, & ! center of thickness categories (m)
+           g_k, & ! probability density function (thickness, 1/m)
+           P_x    ! probability for each thickness category
+
+  real, dimension(ncat_b) :: & ! normal dist for bathymetry
+           y_n, & ! center of bathymetry categories (m)
+           b_n, & ! probability density function (bathymetry, 1/m)
+           P_y    ! probability for each bathymetry category
+
+  integer, dimension(ncat_b) :: tmp
+  real, dimension(0:IG%CatIce) :: hin_max   ! category limits (m)
+
+  logical, dimension (ncat_b) :: gt
+
+  real, dimension(IG%CatIce) :: vcat   ! category limits (m)
+  real, dimension(IG%CatIce) :: acat   ! category limits (m)
+
+  real :: wid_i, wid_b, mu_i, sigma_i, mu_b, m_i, v_i !  parameters for PDFs
```

Comment:
Please add comments describing the purpose and units of each of these real variables.